### PR TITLE
Epoch callback for offline .fit method

### PR DESCRIPTION
**Pull request by joshuaspear**
Added callback post epoch to the LearnableBase fit method

### CODE REVIEW

**Comment by joshuaspear** [comment]
@takuseno, looks like the CI/CD checks are failing due to a failure to import numpy. Any thoughts on this sorry?

File: d3rlpy/base.py [hunk 1]
```diff
@@ -480,7 +487,9 @@ def fitter(
             shuffle: flag to shuffle transitions on each epoch.
             callback: callable function that takes ``(algo, epoch, total_step)``
                 , which is called every step.
-
+            epoch_callback: callable function that takes 
+                ``(algo, epoch, total_step)``, which is called every epoch.
```
Comment (takuseno) [inline d3rlpy/base.py:491]: I'd like to explicitly explain it's called at the end of epochs.
```
``(algo, epoch, total_step)``, which is called at the end of every epoch.
```

**Review by takuseno — CHANGES_REQUESTED** [review]
Thanks for the PR! I see some coding style issues. Can you try this?
```
pip install black==22.3.0 yapf isort
./scripts/format
```

File: d3rlpy/base.py (hunk 1 above)
Comment (joshuaspear) [reply d3rlpy/base.py:491]: Makes sense - have included

**Comment by joshuaspear** [comment]
> @takuseno I've run the formatting scripts - sorry for missing that! The github action still seems to be failing. Just looking at similar issues (https://stackoverflow.com/questions/60180802/github-python-action-failing-on-numpy) and the .github actions, I wonder if it's worth explicitely pip installing numpy for the format check and macos build? Really not familiar with this kind of stuff so might be barking up the wrong tree!

**Comment by takuseno** [comment]
Thanks for the update! It seems that there is an issue in CI side. But, I locally checked your branch and confirmed it's okay. So let me merge this PR. Thank you for your contribution!

Note that the next major update is under development in [nightly](https://github.com/takuseno/d3rlpy/tree/nightly) branch.

**Review by takuseno — APPROVED** [review]

**Pull request merged**